internal/internal.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+
+	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+)
+
+type ExtensionInterface interface {
+	GetPackageSpec() *ocv1alpha1.ExtensionSourcePackage
+	GetUpgradeConstraintPolicy() ocv1alpha1.UpgradeConstraintPolicy
```

Comment:
Since I started review `GetUpgradeConstraintPolicy` was resolved. So resolving this thread. but I have a suggestion about the whole interface [here](https://github.com/operator-framework/operator-controller/pull/598#discussion_r1491036970).

Reply:
(already outdated!)